bundles/org.openhab.binding.iammeter/src/main/java/org/openhab/binding/iammeter/internal/Iammeter3080THandler.java

```diff
@@ -0,0 +1,153 @@
+            this.refreshJob = refreshJob;
+            updateStatus(ThingStatus.UNKNOWN);
+        }
+    }
+
+    @SuppressWarnings("null")
+    private boolean refresh() {
+        refreshCache.invalidateValue();
+        IammeterConfiguration config = this.config;
+        try {
+            String httpMethod = "GET";
+            String url = "http://" + config.username + ":" + config.password + "@" + config.host + ":" + config.port
+                    + "/monitorjson";
+            String content = "";
+            InputStream stream = new ByteArrayInputStream(content.getBytes(StandardCharsets.UTF_8));
+
+            String response = HttpUtil.executeUrl(httpMethod, url, stream, null, TIMEOUT_MS);
+            JsonElement iammeterDataElement = new JsonParser().parse(response);
+            JsonObject iammeterData = iammeterDataElement.getAsJsonObject();
+            String keyWord = "Datas";
+            if (iammeterData.has("Datas") && iammeterData.has("SN")) {
+                String groups[] = { "powerPhaseA", "powerPhaseB", "powerPhaseC" };
+                for (int row = 0; row < groups.length; row++) {
+                    String gpName = groups[row];
+                    List<Channel> chnList = getThing().getChannelsOfGroup(gpName);
+                    for (IammeterWEM3080Channel channelConfig : IammeterWEM3080Channel.values()) {
+                        Channel chnl = chnList.get(channelConfig.ordinal());
+                        if (chnl != null) {
+                            State state = getDecimal(iammeterData.get(keyWord).getAsJsonArray().get(row)
+                                    .getAsJsonArray().get(channelConfig.ordinal()).toString(), channelConfig.getUnit());
+                            updateState(chnl.getUID(), state);
+                        }
+                    }
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            }
+            stream.close();
+            updateStatus(ThingStatus.ONLINE);
+            return true;
+            // Very rudimentary Exception differentiation
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Communication error with the device: " + e.getMessage());
+        } catch (JsonSyntaxException je) {
+            logger.warn("Invalid JSON when refreshing source {}: {}", getThing().getUID(), je.getMessage());
+        }
+        return false;
+    }
+
+    private State getDecimal(String value, Unit<?> unit) {
+        try {
+            return QuantityType.valueOf(Float.parseFloat(value), unit);
+        } catch (NumberFormatException e) {
+            return UnDefType.UNDEF;
+        }
+    }
+
+    @Override
+    public void dispose() {
+        ScheduledFuture<?> refreshJob = this.refreshJob;
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            refreshJob = null;
```

Comment:
You need to set the field to null:
```suggestion
            this.refreshJob = null;
```

Reply:
changed.